praw/models/reddit/emoji.py

```diff
@@ -0,0 +1,63 @@
+"""Provide the EMOJI class."""
+from ...const import API_PATH
+from .base import RedditBase
+
+
+class Emoji(RedditBase):
+    """An individual Emoji object."""
+
+    __hash__ = RedditBase.__hash__
+    STR_FIELD = 'name'
+
+    @property
+    def __init__(self, reddit, subreddit, name, _data=None):
+        """Construct an instance of the Emoji object."""
+        self.name = name
+        self.subreddit = subreddit
+        super(Emoji, self).__init__(reddit, _data)
+        self._mod = None
+
+    def add(self, filepath):
+        """Add an emoji to this subreddit.
+
+        :param emoji: An emoji name (e.g., ``'cake'``) or
+            :class:`~.Emoji` instance.
+
+        To add ``'cake'`` to the subreddit ``'praw_test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('praw_test').emoji['cake'].add('cake.png')
+
+        """
+        data = None
+        if filepath[-4:].lower() == '.png':
+            data = {'filepath': filepath, 'mimetype': 'image/png'}
+        elif filepath[-4:].lower() == '.jpg' \
+                or filepath[-4:].lower() == '.jpeg':
```

Comment:
For both of these if statements, instead of slicing and comparing, can you please use `.endswith()`? For example, `if filepath.lower().endswith('.png'):`

This way, it will be a little harder to let bugs like `filepath[-4:].lower() == '.jpeg'` (which should be `filepath[-5:].lower() == '.jpeg'`) slip through.